File: Mrh.Messaging/Message.cs
using System;

namespace Mrh.Messaging
{
    /// <summary>
    ///     Represents an unprocessed message.
    /// </summary>
    /// <typeparam name="TPayloadType">The type for the payload.</typeparam>
    /// <typeparam name="TBody">The type for the body.</typeparam>
    public class Message<TPayloadType, TBody> where TPayloadType: struct
    {
        
        /// <summary>
        ///     The application generated request id.
        /// </summary>
        public long RequestId { get; set; }
        
        /// <summary>
        ///     The message identifier which also includes the person sending the message.
        /// </summary>
        public MessageIdentifier MessageIdentifier { get; set; }
        
        /// <summary>
        ///     Used for an even on who to send the message to.
        /// </summary>
        public string ToConnectionId { get; set; }
        
        public MessageType MessageType { get; set; }
        
        /// <summary>
        ///     The result of processing the message.
        /// </summary>
        public MessageResultType MessageResultType { get; set; }
        
        public TBody Body { get; set; }
        
        public TPayloadType PayloadType { get; set; }
        
        public Guid UserId { get; set; }
    }
}

File: typescript/webSocket/src/app/share/Api/Messaging/Message.ts
import {MessageType} from './MessageType';
import {MessageResultType} from './MessageResultType';

export class Message<TPayloadType, TBody> {

  constructor(
    public requestId: number,
    public connectionId: string,
    public correlationId: number,
    public messageType: MessageType,
    public messageResultType: MessageResultType,
    public body: TBody,
    public payloadType: TPayloadType,
    public userId: string) {

  }
}
